cabra/src/pages/components/FilterForm.tsx

```diff
@@ -0,0 +1,162 @@
+import DatePicker from "react-datepicker";
+import { IFilterFormState } from "../../types/filterFormState";
+import { ITag } from "../../types/task";
+import { format } from "date-fns";
+import { getTags } from "../../api/tags";
+import { styled } from "twin.macro";
+import tw from "twin.macro";
+import { useQuery } from "react-query";
+
+type SelectOption = {
+  label: string;
+  value: string;
+};
+
+type Props = {
+  setFilters: (filters: IFilterFormState) => void;
+  filters: IFilterFormState;
+};
+
+export type FilterFormInputs = IFilterFormState;
+
+const selectStyles: StylesConfig<SelectOption, true> = {
+  option: (provided) => ({
+    ...provided,
+    color: "black",
+    padding: 8,
+  }),
+  control: (provided) => ({
+    ...provided,
+    borderRadius: "12px",
+  }),
+};
+
+const mapTagToOption = (tag: ITag) => ({
+  value: tag.name,
+  label: tag.name,
+});
+
+const statusOptions: SelectOption[] = [
+  { value: "done", label: "Done" },
+  { value: "todo", label: "Todo" },
+  { value: "indirectlyDone", label: "Indirectly done" },
+  { value: "none", label: "None" },
+];
+
+const Container = styled.div(
+  tw`p-4 bg-primary rounded-lg text-white space-y-3`
+);
+const InputContainer = styled.div(
+  tw`rounded-lg pl-2 pr-2 pb-1 pt-1 bg-secondary flex items-center space-x-5`
+);
+const Label = styled.label(tw`w-1/12`);
+
+export default function FilterForm({ filters, setFilters }: Props) {
+  const { data: tagsData } = useQuery(getTags.cacheKey, getTags.run);
+
+  const resetFilters = () => {
+    setFilters({
+      tagNames: null,
+      status: null,
+      titlePattern: null,
+      dateFrom: null,
+      dateTo: null,
+    });
+  };
+
+  return (
+    <Container>
+      <div tw="flex flex-row justify-between hover:opacity-95 px-2">
+        <h1 tw="text-xl font-bold">Filtering</h1>
+        <button
+          tw="rounded-2xl bg-red-700 text-white font-bold py-0.5 px-2"
+          onClick={resetFilters}
+        >
+          Reset filters
+        </button>
+      </div>
+      <InputContainer>
+        <Label>Tags:</Label>
+        <Select
+          tw="flex-1"
+          options={tagsData?.data.map(mapTagToOption)}
+          styles={selectStyles}
+          isMulti
+          onChange={(options) =>
+            setFilters({
+              ...filters,
+              tagNames: options.map(({ value }) => value),
+            })
+          }
+        />
+      </InputContainer>
+      <InputContainer>
+        <Label>Status:</Label>
+        <Select
+          tw="flex-1"
+          options={statusOptions}
+          styles={selectStyles}
+          onChange={(option) =>
+            setFilters({
+              ...filters,
+              status: (option as unknown as SelectOption).value,
+            })
+          }
+          value={
+            filters.status
+              ? statusOptions.filter(
+                  (option) => option.value === filters.status
+                )[0]
```

Comment:
`filter` + `[0]` = `find`
```suggestion
              ? statusOptions.find(({value}) => value === filters.status)
```